shared/src/business/useCases/judgeActivityReport/getCountOfCaseDocumentsFiledByJudgesInteractor.ts

```diff
@@ -0,0 +1,84 @@
+import { AggregatedEventCodesType } from '@web-api/persistence/elasticsearch/fetchEventCodesCountForJudges';
+import { InvalidRequest, UnauthorizedError } from '@web-api/errors/errors';
+import {
+  JUDGE_ACTIVITY_REPORT_ORDER_EVENT_CODES,
+  OPINION_EVENT_CODES_WITH_BENCH_OPINION,
+} from '../../entities/EntityConstants';
+import { JudgeActivityReportSearch } from '../../entities/judgeActivityReport/JudgeActivityReportSearch';
+import {
+  ROLE_PERMISSIONS,
+  isAuthorized,
+} from '@shared/authorization/authorizationClientService';
+import { addDocumentTypeToEventCodeAggregation } from './addDocumentTypeToEventCodeAggregation';
+
+export type CaseDocumentsAggregationReturnType = {
+  orders: AggregatedEventCodesType;
+  opinions: AggregatedEventCodesType;
+};
+
+export type JudgeActivityReportFilters = {
+  endDate: string;
+  startDate: string;
+  judgeId?: string;
+  judges: string[];
+};
+
+export const getCountOfCaseDocumentsFiledByJudgesInteractor = async (
+  applicationContext: IApplicationContext,
+  params: JudgeActivityReportFilters,
+): Promise<CaseDocumentsAggregationReturnType> => {
+  const authorizedUser = applicationContext.getCurrentUser();
+
+  if (!isAuthorized(authorizedUser, ROLE_PERMISSIONS.JUDGE_ACTIVITY_REPORT)) {
+    throw new UnauthorizedError('Unauthorized to view Judge Activity Report');
+  }
+
+  const searchEntity = new JudgeActivityReportSearch(params);
+
+  if (!searchEntity.isValid()) {
+    throw new InvalidRequest('The Search Params for judges are invalid');
+  }
+
+  const { aggregations: opinionAggregationCount, total: opinionTotal } =
+    await applicationContext
+      .getPersistenceGateway()
+      .fetchEventCodesCountForJudges({
+        applicationContext,
+        params: {
+          documentEventCodes: OPINION_EVENT_CODES_WITH_BENCH_OPINION,
+          endDate: searchEntity.endDate,
+          judges: searchEntity.judges,
+          startDate: searchEntity.startDate,
+        },
+      });
+
+  const computedAggregatedOpinionEventCodes =
+    addDocumentTypeToEventCodeAggregation(opinionAggregationCount);
+
+  const { aggregations: orderAggregationCount, total: orderTotal } =
+    await applicationContext
+      .getPersistenceGateway()
+      .fetchEventCodesCountForJudges({
+        applicationContext,
+        params: {
+          documentEventCodes: JUDGE_ACTIVITY_REPORT_ORDER_EVENT_CODES,
+          endDate: searchEntity.endDate,
+          judges: searchEntity.judges,
+          startDate: searchEntity.startDate,
+        },
+      });
+
+  const computedAggregatedOrderEventCodes =
+    addDocumentTypeToEventCodeAggregation(orderAggregationCount);
+
```

Comment:
This is an improvement by consolidating this into one interactor! 👏 

I think we can do better by having the client call the endpoint with the event codes upon which it wants to report. So, it might make two calls for the information it needs; or more calls if sometime in the future we need to support different subsets of event codes. Perhaps other UIs will make use of this.

Reply:
No need to perform this refactor now; I will document it and open up a PR after this is merged.

Reply:
Agreed!